examples/configs/trader/sample_mirror.cfg

```diff
@@ -64,3 +64,17 @@ PER_LEVEL_SPREAD=0.005
 #[[EXCHANGE_HEADERS]]
 #HEADER=""
 #VALUE=""
+
+# below is a description of coinbase api headers
+# For more detail view the coinebase Api documents: here https://docs.pro.coinbase.com/#creating-a-request
+# ENDPOINT=https://api.pro.coinbase.com
+# The api key as a string.
+#CB-ACCESS-KEY=""
+#The base64-encoded signature (see Signing a Message).
+#timestamp + Method + requestPath + body
+#REQUESTTIME:=time.Now()
```

Comment:
@gjgollub it looks like `REQUESTTIME` is a dynamic header, i.e. it needs to be set on a per-request basis, is that right?

Reply:
@nikhilsaraf that's right, it has be done within 30 seconds of the request. Should I note that?

Reply:
@gjgollub in that case, we would require a code change to support this header parameter since the values in the config file are only used as static headers.

The call stack of the function `plugins.MakeTradingExchange(...)` shows how we go from the botConfig's `EXCHANGE_HEADERS` into static headers on the ccxtExchange (via calling `makeCcxtExchange()` inside `loadExchanges()`).

The `Ccxt` struct (in `ccxt.go`) holds a `headersMap` which is a `map[string]string`.
We can change `headersMap` to be a `map[string]HeaderFn`, where `HeaderFn` is a `func(string, string, string) string` which takes in a `string` representing the http `method`, `requestPath`, and `body` which then produces a header string as an output. This will convert these _static_ headers to be more _dynamic_. 

---

For the existing headers we can just have a passthrough function which ignores the inputs and returns a static header value for the given header key, i.e. when making the TradingExchange and setting the headers, instead of passing an `api.ExchangeHeader` with a fixed string for `Value`, we wrap it in a passthrough function to keep it consistent. Here's a sample of how that can look:
```
api.ExchangeHeader{
    Header: header.Header,
    Value: makeStaticHeaderFn(header.Value),
}

// makeStaticHeaderFn is a convenience method
makeStaticHeaderFn(value string) HeaderFn {
    fn := func(method string, requestPath string, body string) string {
        return value
    }
    // need to convert to HeaderFn to work as a api.ExchangeHeader.Value
    return HeaderFn(fn)
}
```

For the `CB-ACCESS-SIGN` header we can use a function like so:
```
func getCoinbaseSignatureFn(base64EncodedSigningKey string) HeaderFn {
    base64DecodedSigningKey := base64Decode(base64EncodedSigningKey)

    // return this inline method casted as a HeaderFn to work as a headerValue
    return HeaderFn(func(method string, requestPath string, body string) string {
        payload := fmt.Sprintf("%s%s%s%s", time.Now(), method, requestPath, body)
        signature := signMessage(payload, base64DecodedSigningKey)
        base64EncodedHeader := base64Encode(signature)
        return base64EncodedHeader
    })
}
```
This function will be saved as a pre-defined function in a map, pointed to by the string key `COINBASE_FN_CB-ACCCESS-KEY`. We can assign this function on to the api.ExchangeHeader.Value whenever the config file specifies this function.

When setting the headers via the config we can specify the function to use followed by the required params, so for example:
```
CB-ACCCESS-KEY="STATIC:thisismykey"
CB-ACCCESS-KEY="COINBASE_FN_CB-ACCCESS-KEY:mycoinbasesigningkey"
```

This is a fairly more involved change than you may have anticipated given that it requires a code change and small refactor because of the dynamic nature of one of the Coinbase headers. Let me know if you are still up to making this change and if you have any questions about this. Happy to get on a call if that will be helpful.

Reply:
Sure, I'll give it a try

Reply:
great, thanks!